lapce-proxy/src/terminal.rs

```diff
@@ -307,3 +327,40 @@ fn set_locale_environment() {
         .replace('-', "_");
     std::env::set_var("LC_ALL", locale + ".UTF-8");
 }
+
+#[cfg(target_os = "linux")]
+fn flatpak_get_default_host_shell() -> String {
+    let env_string = Command::new("flatpak-spawn")
+        .arg("--host")
+        .arg("printenv")
+        .output()
+        .unwrap()
```

Comment:
would this panic on non-flatpak?

Reply:
only if `/.flatpak-info` exists, which is unlikely

Reply:
it's only executed in 
```rs
if inside_flatpak {
```

Reply:
As explained in a comment, checking for `/.flatpak-info` is the de-facto way of checking whether you are inside of a Flatpak container. I don't think there's any reasonable scenario where that file would exist outside of a Flatpak container. Furthermore, if you did create that file on your host for whatever reason, you would break more apps than Lapce.

Reply:
That's fine then.

If you could fix the fmt and clippy, it's good to merge.